src/App.js

```diff
@@ -1,45 +1,113 @@
-import logo from './logo.svg';
+import { initializeApp } from "firebase/app";
+import {collection, deleteDoc, doc, getFirestore, query, setDoc, orderBy} from "firebase/firestore";
 import './App.css';
 import './TaskItem';
-import './DataContainer';
-import {useEffect, useState} from "react";
 import AppHeader from "./AppHeader";
 import TaskList from "./TaskList";
+import {useCollectionData} from "react-firebase-hooks/firestore";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {useState} from "react";
 
-function App(props) {
-    const [hideUncompleted, setHideUncompleted] = useState(true);
-    const [displayData, setDisplayData] = useState(props.data);
+const firebaseConfig = {
+    apiKey: "AIzaSyD2Qc2rKu6YBO6pugcmKQ65JQhSS7VlmEQ",
+    authDomain: "cs124-lab-3.firebaseapp.com",
+    projectId: "cs124-lab-3",
+    storageBucket: "cs124-lab-3.appspot.com",
+    messagingSenderId: "958194478289",
+    appId: "1:958194478289:web:7c9e5383b578ce60f429f0"
+};
 
-    useEffect(()=>{setDisplayData(props.data)},[props.data]);
+const firebaseApp = initializeApp(firebaseConfig);
+const db = getFirestore(firebaseApp);
+
+const collectionName = "Task-Items";
+
+function App() {
+    const [hideCompleted, setHideCompleted] = useState(false);
+    const [sortType, setSortType] = useState("priority");
+
+    const [editedID, setEditedID] = useState(null);
+    const q = query(collection(db, collectionName), orderBy(sortType, "asc"));
+
+    const [taskItems, loading, error] = useCollectionData(q);
+    console.log(taskItems);
 
     function handleUncompleted(){
-        if (hideUncompleted) {
-            setDisplayData(props.data.filter(taskItem => !taskItem.isCompleted));
-        }
-        else{
-            setDisplayData(props.data)
-        }
-        setHideUncompleted(!hideUncompleted);
+        setHideCompleted(!hideCompleted);
         console.log(displayData)
     }
 
     function handleDelete(){
-        props.onDataChange(props.data.filter(taskItem => !taskItem.isCompleted));
-        //setDisplayData(props.data);
-        console.log(props.data)
-        console.log(displayData)
+        // props.onDataChange(taskItems.filter(taskItem => !taskItem.isCompleted));
+
+       taskItems.forEach(taskItem => taskItem.isCompleted? deleteDoc(doc(db, collectionName, taskItem.taskId)):taskItem);
+     }
+
+    function handleChange(taskID, field, value) {
+        console.log(taskID, field, value);
+        //setData(data.map(taskItem => taskItem.taskId === taskID ? {...taskItem, [field]:value}:taskItem))
+
+        void setDoc(doc(db, collectionName, taskID),
+            {[field]: value}, {merge: true})
+    }
+
+    function handlePlusClick() {
```

Comment:
This function name is tied to the actual UI. If you called it handleAddEmptyTask, it'd be correct even if you switched to a different method of creating empty tasks.